### PR TITLE
perf(ARM): resource plugins support updateArmTemplates 

### PR DESCRIPTION
**Pull request by wenytang-ms**

### CODE REVIEW

File: packages/fx-core/src/plugins/solution/fx-solution/arm.ts [hunk 1]
```diff
@@ -416,18 +422,41 @@ async function doGenerateArmTemplate(ctx: SolutionContext): Promise<Result<any,
...
+      } else {
+        const msg = format(
+          getStrings().solution.GenerateArmTemplateFailNotice,
+          ctx.projectSettings?.appName
+        );
+        ctx.logProvider?.error(msg);
+        return result;
+      }
+    } else if (pluginWithArm.updateArmTemplates) {
+      const pluginContext = getPluginContext(ctx, pluginWithArm.name);
```
Comment (blackchoey) [inline packages/fx-core/src/plugins/solution/fx-solution/arm.ts:449]: no need to run update arm templates for selected plugin?

File: packages/fx-core/src/plugins/solution/fx-solution/arm.ts [hunk 2]
```diff
@@ -443,7 +472,7 @@ async function doGenerateArmTemplate(ctx: SolutionContext): Promise<Result<any,
 
   // Write bicep content to project folder
   if (bicepOrchestrationTemplate.needsGenerateTemplate()) {
-    await backupExistingFilesIfNecessary(ctx);
+    // await backupExistingFilesIfNecessary(ctx);
```
Comment (blackchoey) [inline packages/fx-core/src/plugins/solution/fx-solution/arm.ts:475]: remove this comment?

File: packages/fx-core/src/plugins/solution/fx-solution/arm.ts [hunk 3]
```diff
@@ -468,18 +497,20 @@ async function doGenerateArmTemplate(ctx: SolutionContext): Promise<Result<any,
...
     }
 
-    await fs.writeFile(
-      path.join(templateFolderPath, bicepOrchestrationProvisionFileName),
-      bicepOrchestrationProvisionContent
-    );
-    const res = bicepOrchestrationTemplate.applyReference(bicepOrchestrationConfigContent);
+    let res = bicepOrchestrationTemplate.applyReference(bicepOrchestrationProvisionContent);
+    await fs.appendFile(path.join(templateFolderPath, bicepOrchestrationProvisionFileName), res);
+    res = bicepOrchestrationTemplate.applyReference(bicepOrchestrationConfigContent);
```
Comment (blackchoey) [inline packages/fx-core/src/plugins/solution/fx-solution/arm.ts:502]: better naming?

File: packages/fx-core/src/plugins/resource/bot/plugin.ts [hunk 4]
```diff
@@ -189,6 +189,22 @@ export class TeamsBotImpl {
...
+    const result: ArmTemplateResult = {
+      Configuration: {
+        Modules: {
+          botConfiguration: await fs.readFile(
+            path.join(bicepTemplateDir, PathInfo.ConfigurationModuleTemplateV2FileName),
+            ConstantString.UTF8Encoding
+          ),
+        },
+      },
+    };
```
Comment (xzf0587) [inline packages/fx-core/src/plugins/resource/bot/plugin.ts:204]: The return of updateArmTemplates is different from generateArmTemplates. If we add orchestration part in this return, what will happen? 
Should we replace ArmTemplateResult with a specified class?

File: packages/fx-core/src/plugins/solution/fx-solution/arm.ts (hunk 1 above)
Comment (wenytang-ms) [reply packages/fx-core/src/plugins/solution/fx-solution/arm.ts:449]: sure

File: packages/fx-core/src/plugins/resource/bot/plugin.ts (hunk 4 above)
Comment (wenytang-ms) [reply packages/fx-core/src/plugins/resource/bot/plugin.ts:204]: updateArmTemplates means will update moduleConfiguration.bicep files if need to regenerate.
if orchestration is needed in updateArmTemplates? if so, the orchestraction will do append files to provision.bicep or config.bicep files. this operation can support on solution as well

File: packages/fx-core/src/plugins/solution/fx-solution/arm.ts (hunk 2 above)
Comment (wenytang-ms) [reply packages/fx-core/src/plugins/solution/fx-solution/arm.ts:475]: sure

**Pull request closed without merging**